examples/capture/main.rs

```diff
@@ -108,15 +127,24 @@ async fn run() {
     }
 
     if let Ok(()) = buffer_future.await {
-        let data = output_buffer.get_mapped_range(0, wgt::BufferSize::WHOLE);
-        let mut png_encoder = png::Encoder::new(File::create("red.png").unwrap(), size, size);
+        let padded_buffer = output_buffer.get_mapped_range(0, wgt::BufferSize::WHOLE);
+
+        let mut png_encoder = png::Encoder::new(File::create("red.png").unwrap(), width as u32, height as u32);
         png_encoder.set_depth(png::BitDepth::Eight);
         png_encoder.set_color(png::ColorType::RGBA);
-        png_encoder
+        let mut png_writer = png_encoder
             .write_header()
             .unwrap()
-            .write_image_data(data)
-            .unwrap();
+            .into_stream_writer_with_size(real_bytes_per_row);
+
+        // from the padded_buffer we write just the real bytes into the image
+        for y in 0..height as usize {
+            let start = y * padded_bytes_per_row;
+            let end = start + real_bytes_per_row;
```

Comment:
Not entirely necessary, but I would add a comment explaining the use of the unpadded bytes here.

Reply:
Not sure what you mean here. Does your comment still apply after the refactor?

Reply:
Seems clear enough to me post refractor.